crates/yas_periphery/src/tests/test_yas_nft_position_manager.cairo

```diff
@@ -0,0 +1,239 @@
+        #[available_gas(200000000)]
+        #[should_panic(expected: ('CONTRACT_NOT_DEPLOYED', 'ENTRYPOINT_FAILED'))]
+        fn test_fails_if_pool_does_not_exist() {
+            let (yas_nft_position_manager, token_0, token_1) = setup();
+
+            let (min_tick, max_tick) = get_min_tick_and_max_tick();
+
+            yas_nft_position_manager
+                .mint(
+                    MintParams {
+                        token_0: token_0.contract_address,
+                        token_1: token_1.contract_address,
+                        fee: fee_amount(FeeAmount::MEDIUM),
+                        recipient: WALLET(),
+                        tick_lower: min_tick,
+                        tick_upper: max_tick,
+                        amount_0_desired: 100,
+                        amount_1_desired: 100,
+                        amount_0_min: 0,
+                        amount_1_min: 0,
+                        deadline: 1
+                    }
+                );
+        }
+
+        #[test]
+        #[available_gas(200000000)]
+        #[should_panic(
+            expected: (
+                'u256_sub Overflow',
+                'ENTRYPOINT_FAILED',
+                'ENTRYPOINT_FAILED',
+                'ENTRYPOINT_FAILED',
+                'ENTRYPOINT_FAILED'
+            )
+        )]
+        fn test_fails_if_cannot_transfer() {
+            let (yas_nft_position_manager, token_0, token_1) = setup();
+
+            let encode_price_sqrt_1_1 = FP64x96Impl::new(79228162514264337593543950336, false);
+
+            yas_nft_position_manager
+                .create_and_initialize_pool_if_necessary(
+                    token_0.contract_address,
+                    token_1.contract_address,
+                    fee_amount(FeeAmount::MEDIUM),
+                    encode_price_sqrt_1_1
+                );
+
+            set_contract_address(WALLET());
+            token_0.approve(yas_nft_position_manager.contract_address, 0);
+
+            let (min_tick, max_tick) = get_min_tick_and_max_tick();
+            yas_nft_position_manager
+                .mint(
+                    MintParams {
+                        token_0: token_0.contract_address,
+                        token_1: token_1.contract_address,
+                        fee: fee_amount(FeeAmount::MEDIUM),
+                        recipient: WALLET(),
+                        tick_lower: min_tick,
+                        tick_upper: max_tick,
+                        amount_0_desired: 100,
+                        amount_1_desired: 100,
+                        amount_0_min: 0,
+                        amount_1_min: 0,
+                        deadline: 1
+                    }
+                );
+        }
+
+        #[test]
+        #[available_gas(200000000)]
+        fn test_creates_a_token() {
+            let (yas_nft_position_manager, token_0, token_1) = setup();
+
+            let encode_price_sqrt_1_1 = FP64x96Impl::new(79228162514264337593543950336, false);
```

Comment:
I chose this number because it is the one used in the test case for [Uniswap Periphery](https://github.com/Uniswap/v3-periphery/blob/697c2474757ea89fec12a4e6db16a574fe259610/test/NonfungiblePositionManager.spec.ts#L218). We already have a function that returns this value and explains where it comes from

Reply:
✔️